### DISCUSSION TITLE
Welcome to libreswan Discussions!

### DISCUSSION THREAD

**Discussion by libreswan** [libreswan/libreswan#458] [Announcements]
## 👋 Welcome!
  We’re testing out Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by blackmlm** [comment]
Hi,
I have a Rocky Linux 8, with Libreswan 4.5 (XFRM) on 4.18.0-425.3.1.el8.x86_64.

My tunnel is active and  I can ping the SQL server on the other side. I can run run queries, but when the query is too big it just times out.

I have a similar machine behind a sonicwall VPN and the long Query works OK.

I say long is not on return data, but long on query structure.... 
Sample of not working query:
Select
    Documents.DocYYMM,
    OrdersDtl.ItemNumber,
    OrdersDtl.Description,
    OrdersDtl.EnteredQty,
    OrdersDtl.EnteredQtyUM,
    OrdersDtl.EnteredPrice,
    OrdersDtl.EnteredPriceUM,
    OrdersHdr.Account,
    CustomerMaster.CustomerName,
    CustomerMaster.CustomerCompany,
    CustomerMaster.CustomerAddressLine1,
    CustomerMaster.CustomerAddressLine2,
    CustomerMaster.City,
    CustomerMaster.StateCode,
    CustomerMaster.Zip,
    OrdersHdr.SalesBranch,
    OrdersHdr.ExpireDate,
    OrdersHdr.OrderDate,
    OrdersHdr.Subtotal,
    OrdersHdr.SalesTax,
    OrdersHdr.Job,
    OrdersHdr.OrderUser,
    OrdersHdr.OrderTotal
    From
    OrdersHdr with(nolock) Inner Join
    Documents with(nolock) On OrdersHdr.DocIDInternal = Documents.DocIDInternal Inner Join
    OrdersDtl with(nolock) On OrdersDtl.DocIDInternal = OrdersHdr.DocIDInternal Inner Join
    CustomerMaster with(nolock) On OrdersHdr.Account = CustomerMaster.AccountNum
Where
    Documents.DocYYMM = '2302'


Sample of Working Query:

Select
    Documents.DocYYMM,
    OrdersDtl.ItemNumber,
    OrdersDtl.Description,
    OrdersHdr.OrderTotal
    From
    OrdersHdr with(nolock) Inner Join
    Documents with(nolock) On OrdersHdr.DocIDInternal = Documents.DocIDInternal Inner Join
    OrdersDtl with(nolock) On OrdersDtl.DocIDInternal = OrdersHdr.DocIDInternal Inner Join
    CustomerMaster with(nolock) On OrdersHdr.Account = CustomerMaster.AccountNum
Where
    Documents.DocYYMM = '2302'


As you can see is related to the amount of data sent.

any ideas on why?

**Comment by paulwouters** [comment]
If you can ssh in and that works but typing “dmesg” hangs, it’s most likely an MTU issue. See the libreswan FAQ entries on MTU and  tcp clamping Sent using a virtual keyboard on a phoneOn Feb 24, 2023, at 17:21, blackmlm ***@***.***> wrote:﻿
Hi,
I have a Rocky Linux 8, with Libreswan 4.5 (XFRM) on 4.18.0-425.3.1.el8.x86_64.
My tunnel is active and  I can ping the SQL server on the other side. I can run run queries, but when the query is too big it just times out.
I have a similar machine behind a sonicwall VPN and the long Query works OK.
I say long is not on return data, but long on query structure....
Sample of not working query:
Select
Documents.DocYYMM,
OrdersDtl.ItemNumber,
OrdersDtl.Description,
OrdersDtl.EnteredQty,
OrdersDtl.EnteredQtyUM,
OrdersDtl.EnteredPrice,
OrdersDtl.EnteredPriceUM,
OrdersHdr.Account,
CustomerMaster.CustomerName,
CustomerMaster.CustomerCompany,
CustomerMaster.CustomerAddressLine1,
CustomerMaster.CustomerAddressLine2,
CustomerMaster.City,
CustomerMaster.StateCode,
CustomerMaster.Zip,
OrdersHdr.SalesBranch,
OrdersHdr.ExpireDate,
OrdersHdr.OrderDate,
OrdersHdr.Subtotal,
OrdersHdr.SalesTax,
OrdersHdr.Job,
OrdersHdr.OrderUser,
OrdersHdr.OrderTotal
From
OrdersHdr with(nolock) Inner Join
Documents with(nolock) On OrdersHdr.DocIDInternal = Documents.DocIDInternal Inner Join
OrdersDtl with(nolock) On OrdersDtl.DocIDInternal = OrdersHdr.DocIDInternal Inner Join
CustomerMaster with(nolock) On OrdersHdr.Account = CustomerMaster.AccountNum
Where
Documents.DocYYMM = '2302'
Sample of Working Query:
Select
Documents.DocYYMM,
OrdersDtl.ItemNumber,
OrdersDtl.Description,
OrdersHdr.OrderTotal
From
OrdersHdr with(nolock) Inner Join
Documents with(nolock) On OrdersHdr.DocIDInternal = Documents.DocIDInternal Inner Join
OrdersDtl with(nolock) On OrdersDtl.DocIDInternal = OrdersHdr.DocIDInternal Inner Join
CustomerMaster with(nolock) On OrdersHdr.Account = CustomerMaster.AccountNum
Where
Documents.DocYYMM = '2302'
As you can see is related to the amount of data sent.
any ideas on why?

**Reply by blackmlm** [reply]
the problem was with the MTU and the solution within the FAQ.

iptables -I FORWARD -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1380

**Comment by paulwouters** [comment]
Awesome,We hope with IPTFS support in the Linux kernel, that these kind of workarounds won’t be requires anymore in the near future.Sent using a virtual keyboard on a phoneOn Feb 27, 2023, at 17:41, blackmlm ***@***.***> wrote:﻿
the problem was with the MTU and the solution within the FAQ.
iptables -I FORWARD -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1380
